src/plugins/condition/Condition.js

```diff
@@ -189,26 +191,22 @@ export default class ConditionClass extends EventEmitter {
 
     handleCriterionResult(eventData) {
         let id = eventData.id;
-        let result = eventData.data.result;
-        let found = this.findCriterion(id);
-        if (found) {
-            this.criteriaResults[id] = result;
+        if (this.findCriterion(id)) {
+            this.criteriaResults[id] = eventData.data.result;
         }
-        this.handleConditionUpdated();
+        // change criterion result to be condition result as passed down the line
+        // hacky but trying not to make sweeping changes to current architecture
+        eventData.data.result = computeCondition(this.criteriaResults, this.trigger === TRIGGER.ALL);
```

Comment:
Regarding `eventData.data.result = `
I would advise against changing the value of parameters passed into a function. Is there a reason you don't want to create a new variable for this?

Reply:
Made the change